### PR TITLE
Add -Top parameter to Get-AzTableRow commands

### PR DESCRIPTION
**Pull request by backerman**
Same idea as #56 but only adds the one parameter and doesn't change behavior if it's not passed. (This branch has been sitting around for a while but I hadn't actually done the PR yet.)

### CODE REVIEW

File: AzureRmStorageTableCoreHelper.psm1 [hunk 1]
```diff
@@ -56,6 +56,19 @@ function ExecuteQueryAsync
 			$Results = $Table.ExecuteQuerySegmentedAsync($TableQuery, $token)
 			$token = $Results.Result.ContinuationToken
 			$AllRows += $Results.Result.Results
+			# TakeCount controls the number of results returned per page, not
+			# for the entire query. See e.g. the note in
+			# https://docs.microsoft.com/azure/cosmos-db/table-storage-design-guide#retrieve-large-numbers-of-entities-from-a-query
+			if (($null -ne $token) -and ($null -ne $TableQuery.TakeCount)) {
```
Comment (paulomarquesc) [inline AzureRmStorageTableCoreHelper.psm1]: This seems good but please let's follow the formatting of most of the code:

```powershell
			if (($null -ne $token) -and ($null -ne $TableQuery.TakeCount))
                        {
				# If the take count is larger than the number of rows in this
				# segment, there are more rows to return.
				if ($TableQuery.TakeCount -gt $Results.Result.Results.Count)
                                {
					$TableQuery.TakeCount -= $Results.Result.Results.Count
				}
                                else
                                {
					# No more rows are available in the current page.
					break
				}
			}
```

File: AzureRmStorageTableCoreHelper.psm1 [hunk 2]
```diff
@@ -306,13 +321,16 @@ function Get-AzTableRowByPartitionKey
 
 		[Parameter(Mandatory=$true)]
 		[AllowEmptyString()]
-		[string]$PartitionKey
+		[string]$PartitionKey,
+
+		[Parameter()]
```
Comment (paulomarquesc) [inline AzureRmStorageTableCoreHelper.psm1]:
```suggestion
		[[Parameter(Mandatory=$false)]]

```

File: AzureRmStorageTableCoreHelper.psm1 [hunk 3]
```diff
@@ -349,14 +369,17 @@ function Get-AzTableRowByPartitionKeyRowKey
 
 		[Parameter(Mandatory=$true)]
 		[AllowEmptyString()]
-		[string]$RowKey
+		[string]$RowKey,
+
+		[Parameter()]
```
Comment (paulomarquesc) [inline AzureRmStorageTableCoreHelper.psm1]:
```suggestion
		[[Parameter(Mandatory=$false)]]

```

File: AzureRmStorageTableCoreHelper.psm1 [hunk 4]
```diff
@@ -403,7 +428,10 @@ function Get-AzTableRowByColumnName
 
 		[Parameter(Mandatory=$true)]
 		[validateSet("Equal","GreaterThan","GreaterThanOrEqual","LessThan" ,"LessThanOrEqual" ,"NotEqual")]
-		[string]$Operator
+		[string]$Operator,
+
+		[Parameter()]
```
Comment (paulomarquesc) [inline AzureRmStorageTableCoreHelper.psm1]:
```suggestion
		[[Parameter(Mandatory=$false)]]

```

File: AzureRmStorageTableCoreHelper.psm1 [hunk 5]
```diff
@@ -452,14 +482,17 @@ function Get-AzTableRowByCustomFilter
 		$Table,
 
 		[Parameter(Mandatory=$true)]
-		[string]$CustomFilter
+		[string]$CustomFilter,
+
+		[Parameter()]
```
Comment (paulomarquesc) [inline AzureRmStorageTableCoreHelper.psm1]:
```suggestion
		[[Parameter(Mandatory=$false)]]

```

File: AzureRmStorageTableCoreHelper.psm1 [hunk 6]
```diff
@@ -546,7 +581,10 @@ function Get-AzTableRow
 		[string]$Operator,
 		
 		[Parameter(Mandatory=$true, ParameterSetName="byCustomFilter")]
-		[string]$CustomFilter
+		[string]$CustomFilter,
+
+		[Parameter()]
```
Comment (paulomarquesc) [inline AzureRmStorageTableCoreHelper.psm1]:
```suggestion
		[[Parameter(Mandatory=$false)]]

```

File: AzureRmStorageTableCoreHelper.psm1 [hunk 7]
```diff
@@ -595,6 +633,9 @@ function Get-AzTableRow
 		$TableQuery.FilterString = $Filter
 	}
 
+	# Set number of rows to return ($Top is null if unset, in which case there's no limit.)
+	$TableQuery.TakeCount = $Top
```
Comment (paulomarquesc) [inline AzureRmStorageTableCoreHelper.psm1]: Please add a null check before assigning the TakeCount property.

**Review by paulomarquesc — COMMENTED** [review]
Hi @beckerman, thanks for the PR. I included a few comments, plus the documentation (markdown files per cmdlet) needs to be updated to include the new option.

File: AzureRmStorageTableCoreHelper.psm1 (hunk 1 above)
Comment (backerman) [reply AzureRmStorageTableCoreHelper.psm1]: Fixed.

File: AzureRmStorageTableCoreHelper.psm1 (hunk 2 above)
Comment (backerman) [reply AzureRmStorageTableCoreHelper.psm1]: Fixed.

**Comment by backerman** [comment]
Applied all changes (before noticing GIthHub now has a one-click button to commit them). Working on Markdown now.

File: AzureRmStorageTableCoreHelper.psm1 (hunk 7 above)
Comment (backerman) [reply AzureRmStorageTableCoreHelper.psm1]: Done.

**Comment by backerman** [comment]
Markdown completed - ended up adding missing `-SelectColumn` parameters since it was already in the platyPS output.

**Comment by backerman** [comment]
Also, was I supposed to be pre-squashing the commits on this PR?

**Comment by paulomarquesc** [comment]
Thanks @backerman , this looks good now. I executed the unit tests locally here and all passed:

image

**Comment by paulomarquesc** [comment]
Published on PSGallery as version 2.1.0.

**Pull request merged**